src/fiatconnect/FiatDetailsScreen.tsx

```diff
@@ -0,0 +1,306 @@
+import {
+  FiatAccountSchema,
+  FiatAccountType,
+  FiatConnectError,
+} from '@fiatconnect/fiatconnect-types'
+import { StackScreenProps } from '@react-navigation/stack'
+import React, { useLayoutEffect, useMemo, useRef, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Image, ScrollView, StyleSheet, Text, View } from 'react-native'
+import { SafeAreaView } from 'react-native-safe-area-context'
+import { useDispatch } from 'react-redux'
+import { showError, showMessage } from 'src/alert/actions'
+import { ErrorMessages } from 'src/app/ErrorMessages'
+import BorderlessButton from 'src/components/BorderlessButton'
+import Button, { BtnSizes } from 'src/components/Button'
+import TextInput from 'src/components/TextInput'
+import { addNewFiatAccount } from 'src/fiatconnect'
+import i18n from 'src/i18n'
+import ForwardChevron from 'src/icons/ForwardChevron'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import { StackParamList } from 'src/navigator/types'
+import { userLocationDataSelector } from 'src/networkInfo/selectors'
+import useSelector from 'src/redux/useSelector'
+import colors from 'src/styles/colors'
+import fontStyles from 'src/styles/fonts'
+import { Currency } from 'src/utils/currencies'
+import Logger from 'src/utils/Logger'
+
+const TAG = 'FiatDetailsScreen'
+type ScreenProps = StackScreenProps<StackParamList, Screens.FiatDetailsScreen>
+
+type Props = ScreenProps
+
+interface Fields {
+  name: string
+  label: string
+  regex: RegExp
+  placeholderText: string
+  errorMessage: string
+}
+
+// This is a mapping between different fiat account schema to the metadata of the fields that need to be rendered on the bank details screen
+const SCHEMA_TO_FIELD_METADATA_MAP = {
```

Comment:
This also doesn't take advantage of many of the Types we have for FiatConnect, and makes it so that the params for a given Schema are spread out in this const and implicitFields(). It would be nice if we could combine them both in a data structure.

While thinking through this, I went a bit ham and almost fully implemented a new structure. let me know what you think

```ts
interface FormParameter {
  label: string
  regex: RegExp
  placeholderText: string
  errorMessage: string
}
interface ImplicitParameter<T> {
  value: T
}

interface SchemaMetadata {
  [FiatAccountSchema.AccountNumber]: {
    [Property in keyof AccountNumber]: FormParameter | ImplicitParameter<AccountNumber[Property]>
  }
}

const getSchemaMetadata = (country: string, fiatAccountType: FiatAccountType): SchemaMetadata => ({
  [FiatAccountSchema.AccountNumber]: {
    accountName: {
      label: i18n.t('fiatAccountSchema.accountName.label'),
      regex: /.*?/,
      placeholderText: i18n.t('fiatAccountSchema.accountName.placeholderText'),
      errorMessage: i18n.t('fiatAccountSchema.accountName.errorMessage'),
    },
    institutionName: {
      label: i18n.t('fiatAccountSchema.institutionName.label'),
      regex: /.*?/,
      placeholderText: i18n.t('fiatAccountSchema.institutionName.placeholderText'),
      errorMessage: i18n.t('fiatAccountSchema.institutionName.errorMessage'),
    },
    accountNumber: {
      label: i18n.t('fiatAccountSchema.accountNumber.label'),
      regex: /^[0-9]{10}$/,
      placeholderText: i18n.t('fiatAccountSchema.accountNumber.placeholderText'),
      errorMessage: i18n.t('fiatAccountSchema.accountNumber.errorMessage'),
    },
    country: { value: country},
    fiatAccountType: { value: fiatAccountType}
  }
})
```